usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/GenerateExclusionFilenameListTest.java

```diff
@@ -16,261 +16,191 @@
+        assertEquals(0, toCheck.size());
+    }
+
+    @Test
+    void excludeFileWithDepth0ShouldExcludeAllFile() {
+        final GtfsNode level2OnlyChild = spy(new GtfsNode("level2_only_child", null));
+        final GtfsNode level1SecondChild = spy(new GtfsNode("level1_second_child", null));
+        final GtfsNode level1FirstChild = spy(
+                new GtfsNode("level1_first_child", new ArrayList<>(List.of(level2OnlyChild))));
+        final GtfsNode level0Node = spy(
+                new GtfsNode("level0_file", new ArrayList<>(List.of(level1FirstChild, level1SecondChild))));
+
+        final List<GtfsNode> childrenCollection =
+                new ArrayList<>(List.of(level0Node, level1FirstChild, level1SecondChild, level2OnlyChild));
+        final GtfsNode mockRoot = spy(new GtfsNode("level0_file", childrenCollection));
+
+        final GtfsSpecRepository mockGtfsSpecRepo = spy(GtfsSpecRepository.class);
+        when(mockGtfsSpecRepo.getOptionalFilenameList()).thenReturn(Collections.emptyList());
+        when(mockGtfsSpecRepo.getRequiredFilenameList())
+                .thenReturn(List.of("level0_file", "level1_first_child", "level1_second_child", "level2_only_child"));
+
+        final ExecParamRepository mockExecParamRepo = mock(ExecParamRepository.class);
+        when(mockExecParamRepo.getExecParamValue(ExecParamRepository.EXCLUSION_KEY))
+                .thenReturn("[level0_file]");
+        when(mockGtfsSpecRepo.getGtfsRelationshipDescriptor()).thenReturn(mockRoot);
+
+        final Logger mockLogger = mock(Logger.class);
+
         final GenerateExclusionFilenameList underTest =
-                new GenerateExclusionFilenameList(mockGtfsSpecRepo, mockExecParamRepo);
-        final Exception exception = assertThrows(IllegalArgumentException.class, underTest::execute);
+                new GenerateExclusionFilenameList(mockGtfsSpecRepo, mockExecParamRepo, mockLogger);
+
+        final ArrayList<String> toCheck = underTest.execute();
+
+        verify(mockExecParamRepo, times(1))
+                .getExecParamValue(ExecParamRepository.EXCLUSION_KEY);
+        verify(mockGtfsSpecRepo, times(1)).getOptionalFilenameList();
+        verify(mockGtfsSpecRepo, times(1)).getRequiredFilenameList();
+        verify(mockGtfsSpecRepo, times(1)).getGtfsRelationshipDescriptor();
+        verify(level0Node, times(1)).DFS(any());
+
+        assertEquals(4, toCheck.size());
+        assertTrue(toCheck.contains("level0_file"));
+        assertTrue(toCheck.contains("level1_first_child"));
+        assertTrue(toCheck.contains("level1_second_child"));
+        assertTrue(toCheck.contains("level2_only_child"));
+
+        verify(mockRoot, times(1)).getChildByName(ArgumentMatchers.eq("level0_file"));
+        verify(mockRoot, times(1)).getChildren();
+
+        verifyNoMoreInteractions(mockExecParamRepo, mockGtfsSpecRepo, mockRoot);
+    }
+
+    @Test
+    void excludeAllFileWithDepth1ShouldExclude3File() {
```

Comment:
We could benefit from a small visual representation of the graph under test in the form of comments or javadoc